utils/utils.go

```diff
@@ -0,0 +1,189 @@
+package utils
+
+// #include "limits.h"
+import "C"
+
+import (
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gluster/glusterd2/errors"
+)
+
+// IsLocalAddress checks whether a given host/IP is local
+func IsLocalAddress(host string) bool {
+	var finalHost string
+	found := false
+
+	addrs, e := net.LookupHost(host)
+	if e != nil {
+		log.Error(e.Error())
+		return found
+	}
+	for _, addr := range addrs {
+		names, err := net.LookupAddr(addr)
+		if err != nil {
+			log.Error(e.Error())
+			return found
+		}
+		for _, name := range names {
+			// LookupHost/LookupAddr appends a '.' at the end of the
+			// hostname, not sure why though, so this is hack as of
+			// now
+			finalHost = name[0 : len(name)-1]
+			if finalHost == host ||
+				finalHost == "localhost.localdomain" ||
+				finalHost == "localhost" {
+				found = true
+				return found
+			}
+		}
+	}
+	return found
+}
+
+// ParseHostAndBrickPath parses the host & brick path out of req.Bricks list
+func ParseHostAndBrickPath(brickPath string) (string, string) {
+	idx := strings.LastIndex(brickPath, ":")
+	if idx == -1 {
+		log.Error("Invalid brick path, it should be in the form of host:path")
+		return "", ""
+	}
+	hostname := brickPath[0:idx]
+	path := brickPath[idx+1 : len(brickPath)]
+
+	return hostname, path
+}
+
+//ValidateBrickPathLength validates the length of the brick path
+func ValidateBrickPathLength(brickPath string) int {
```

Comment:
No, this is what I already discussed with you. Even if OS does validate about the PATH_MAX but it will create the subsequent sub directories while attempting a mkdir which we don't want.